.github/workflows/rust-lint.yml

```diff
@@ -36,4 +36,8 @@ jobs:
     - name: Check code formatting
       run: cargo +nightly fmt --all -- --check
     - name: Check cargo clippy warnings
+      run: cargo clippy --workspace -- -D warnings
+    - name: Check cargo clippy warnings for all targets
+      run: cargo clippy --workspace --all-targets -- -D warnings
+    - name: Check cargo clippy warnings for all targets and features
       run: cargo clippy --workspace --all-targets --all-features -- -D warnings
```

Comment:
Do we need all three of them? Doesn't `--workspace --all-targets` *strictly* supersedes `--workspace`? 🤔

Reply:
Step without `--all-features` allows us to find places where we try to use testing code in a non-testing environment. Clippy may fail in this case by non-existsting entities

Reply:
> Step without `--all-features`...

Yeah, this one I get. I'm asking about `--workspace` vs  `--workspace --all-targets`. It seems to me, like `--workspace` without `--all-targets` is not required.

Reply:
> --all-targets
> Check all targets. This is equivalent to specifying --lib --bins --tests --benches --examples.

Apparently, it enables features from `[dev-dependencies]` not just only for `--tests` and `--benches`, but also for `--lib` and `--bins`. I've added it to check that `--lib`/`--bins` compile with this feature disabled.

Reply:
Ok, got it. Thanks.